### PR TITLE
Update amend add content to reference pages in SDK docs - closes #550

### LINKED ISSUE #550: Update, amend and add content to reference pages in SDK docs
Reference: add short descriptions for the following: Commander, Commands, Core commands
Reference>Developer changelog: New title.
Reference>Lisk Elements: In the package list under Specification - remove 'docs' after each package name.
Reference>Lisk Elements>API client>DApps: Does it still exist - check with Devs if deactivated, if so then remove.
Reference>Lisk Elements>Client. use 'packages'. This provides a default set of elements for use.

### PR DESCRIPTION
**Pull request by chris529**
closes #550 

parent issue #535

### CODE REVIEW

File: modules/ROOT/pages/reference/index.adoc [hunk 1]
```diff
@@ -23,29 +24,73 @@ Mona Bärenfänger <mona@lightcurve.io>
...
 
-The reference section is designed to provide the user with a comprehensive description of all the necessary specifications, including information for the Lisk elements xref:{url_lisk_packages}[packages], together with the command references for the xref:{url_lisk_commander}[Lisk Commander].
-
-The references for the Lisk Commander are broken down into the following subsections listed below, in order to assist the user in locating the relevant reference information required:
+The references section is designed to provide the user with a comprehensive description of all the necessary specifications.
+A brief description of the contents of the references are listed below, in order to assist the user in locating the relevant reference information required.
 
 *  xref:{url_commander_overview}[Commander]
+
+Lisk Commander is an effective, convenient and user-friendly tool designed for all levels of developers, and further useful information is included within the reference section to assist the user.
```
Comment (Tschakki) [inline modules/ROOT/pages/reference/index.adoc]: I think at this point we should not explain what Lisk Commander is, as we do this already on the SDK and Commander overview page, but rather give a quick summary of what content is to find in the Lisk Commander reference.

**Review by Tschakki — CHANGES_REQUESTED** [review]
Please have a look at the reference overview again.

I notice you explain briefly, what every software tool is.
As this should be already be explained in other parts of the documentation, it is not necessary to explain it here again. Where adequate, you can instead link to the existing content.

Just focus on what content they can find in the reference pages and try to keep it as short as possible, when describing single pages.

Also, please make a decision if you want to make a list of every existing page in references (as already mentioned I wouldn't favor that, because it is hard to maintain) or rather have a summary with the most important links and explanations about what content to find in this section.

If you want to go for a full list, please find a way to make it clear which pages are child pages of other pages, and also ensure to not only include links to lisk commander and lisk elements, as the reference section has more pages, like the config reference, the api reference and the developer changelog. In a summary, these 3 should of course be mentioned as well,  because they are quite important.

File: modules/ROOT/pages/reference/index.adoc (hunk 1 above)
Comment (chris529) [reply modules/ROOT/pages/reference/index.adoc]: Agreed.

File: modules/ROOT/pages/reference/index.adoc [hunk 2]
```diff
@@ -22,30 +23,52 @@ Mona Bärenfänger <mona@lightcurve.io>
...
+A brief overview of the applicable references are listed below, in order to assist the user in locating the relevant information required.
 
-*  xref:{url_commander_overview}[Commander]
-*  xref:{url_lisk_commander_commands}[Commands]
-*  xref:{url_lisk_core}[Core commands]
-*  xref:{url_sens_inputs}[Sensitive inputs]
+Firstly the  xref:{url_api_specification}[API specification] references provide an in-depth definition of the specifications, followed by the  xref:{url_config}[Configuration] which contains a detailed example explaining the default values. This is followed by the xref:{url_changelog}[Developer changelog] which encompasses an extensive description of the updated database and api changes.
+
+The Lisk xref:{url_commander_overview}[Commander] reference section comprises of the setup, installation and configuration. In addition the following  xref:{url_lisk_commander_commands}[Commands], xref:{url_lisk_core}[Core commands] and xref:{url_sens_inputs}[Sensitive inputs] are also covered separately, together with useful examples in order to assist the end-user.
```
Comment (Tschakki) [inline modules/ROOT/pages/reference/index.adoc]:
```suggestion
The xref:{url_commander_overview}[Lisk Commander] reference section comprises of the setup, installation and configuration. In addition the following  xref:{url_lisk_commander_commands}[Commands], xref:{url_lisk_core}[Core commands] and xref:{url_sens_inputs}[Sensitive inputs] are also covered separately, together with useful examples in order to assist the end-user.
```

File: modules/ROOT/pages/reference/index.adoc [hunk 3]
```diff
@@ -22,30 +23,52 @@ Mona Bärenfänger <mona@lightcurve.io>
...
 
-The references for the Lisk Commander are broken down into the following subsections listed below, in order to assist the user in locating the relevant reference information required:
+The references section is designed to provide the user with a comprehensive description of all the necessary specifications.
+A brief overview of the applicable references are listed below, in order to assist the user in locating the relevant information required.
 
-*  xref:{url_commander_overview}[Commander]
-*  xref:{url_lisk_commander_commands}[Commands]
-*  xref:{url_lisk_core}[Core commands]
-*  xref:{url_sens_inputs}[Sensitive inputs]
+Firstly the  xref:{url_api_specification}[API specification] references provide an in-depth definition of the specifications, followed by the  xref:{url_config}[Configuration] which contains a detailed example explaining the default values. This is followed by the xref:{url_changelog}[Developer changelog] which encompasses an extensive description of the updated database and api changes.
```
Comment (Tschakki) [inline modules/ROOT/pages/reference/index.adoc]:
```suggestion
Firstly the  xref:{url_api_specification}[API specification] provides an in-depth definition of the API endpoints, followed by the  xref:{url_config}[SDK Configuration reference] which contains a detailed example explaining the default values. This is followed by the xref:{url_changelog}[Developer changelog] which encompasses an extensive description of the updated database and api changes that are relevant for developers when updating their application to the latest version of the Lisk SDK.
```

**Review by Tschakki — CHANGES_REQUESTED** [review]
Almost good! Just left 2 suggestions :)

**Comment by chris529** [comment]
Agreed, thanks Mona!

**Review by Tschakki — APPROVED** [review]

**Pull request merged**